system/Pager/PagerRenderer.php

```diff
@@ -117,7 +117,7 @@ public function getPrevious()
 
 		$uri = clone $this->uri;
 
-		$uri->addQuery('page', $this->first - 1);
+		$uri->addQuery('page', $this->current - 1);
```

Comment:
I'm not sure why you're saying there's no example of handling paging and the logic doesn't work. Just because it's working a different way than you seem to prefer doesn't mean it doesn't work. Please provide concrete examples of why it's wrong and what your desired outcome is.